device-src/s3.c

```diff
@@ -2491,11 +2503,19 @@ perform_request(S3Handle *hdl,
     }
 
     if (hdl->s3_api == S3_API_AWS4) {
-	if (read_data) {
-	    data_SHA256Hash = s3_compute_sha256_hash_ba(read_data);
-	} else {
-	    data_SHA256Hash = s3_compute_sha256_hash((unsigned char *)"", 0);
-	}
+        if (read_data) {
+            data_SHA256Hash = s3_compute_sha256_hash_ba(read_data);
+        } else {
+            data_SHA256Hash = s3_compute_sha256_hash((unsigned char *)"", 0);
+        }
+        if (subresource && strcmp(subresource, "lifecycle") == 0 && md5_func) {
+            md5_hash = md5_func(read_data);
+            if (md5_hash) {
+                md5_hash_b64 = s3_base64_encode(md5_hash);
```

Comment:
does this need to be freed?

Reply:
If you mean md5_hash_b64 .. it’s cleaned up at line 2764 along with many other things.

<copying into context>